src/library_glfw.js

```diff
@@ -981,14 +981,9 @@ var LibraryGLFW = {
       if (!win) return;
 
       if (GLFW.active.id == win.id) {
```

Comment:
What about the `Browser.exitFullscreen();` thing... should explictly setting the windows size also not exit fullscreen?

Reply:
That is a good question. SDL does not do it. glut does it. In my opinion SDL is right. When `Browser.requestFullscreen` is called, a `fullscreenchange` listener is put in place to detect exiting full screen (the user pressing ESC, or whatever key triggers that).

Calling `Browser.exitFullscreen();` might trigger an error for the same reason that `Browser.requestFullscreen` can (if the code calls `glfwSetWindowSize` while in fullscreen mode but not from a user triggered event).

I believe the proper way for the client code to handle this is to set a `Module.onFullScreen` callback which gets called whenever the browser goes in/out of fullscreen and then call `glfwSetWindowSize`.

I actually used this technique in my code this morning:


```cpp
    EM_ASM(
        {
            RAYLIB = {}; // used to store width/height before fullscreen
            const cb = Module.requestFullscreen;
            Module.requestFullscreen = (pointerLock, resize) => {
                const canvas = document.getElementById('canvas');
                RAYLIB.screenWidth = canvas.clientWidth;
                RAYLIB.screenHeight = canvas.clientHeight;
                cb(pointerLock, resize);
            };
            Module.onFullscreen = Module.cwrap('FullscreenCallback', null, ['boolean']); // called by emscripten
        });

void FullscreenCallback(bool isFullscreen) {
    if(isFullscreen) {
       // ...
    }
    else  {
        glfwSetWindowSize(EM_ASM_INT(return RAYLIB.screenWidth;), EM_ASM_INT(return RAYLIB.screenHeight;));
    }
}
```

That being said, if you prefer to leave it, I believe the chances of getting an error are lower and probably have less consequences.

Reply:
No, I think being consistent with SDL and glut makes more sense.